### PR TITLE
Toggle enterprise feature

### PR DESCRIPTION
**Pull request by redsk**
## What changes were proposed in this pull request?

Add a variable to cp-kafka/values.yaml to enable or disable enterprise
features. Same as https://github.com/confluentinc/cp-helm-charts/pull/436 .

Fixes #377

This PR is based on the work of @oscarh on #378.

## How was this patch tested?
```
helm template --set cp-kafka.enableEnterpriseFeatures=false . > /tmp/kafka_enterprise_disabled.yaml
helm template --set cp-kafka.enableEnterpriseFeatures=true . > /tmp/kafka_enterprise_enabled.yaml
diff -u /tmp/kafka_enterprise_{enabled,disabled}.yaml
```
```
--- /tmp/kafka_enterprise_enabled.yaml	2021-12-14 20:51:50.000000000 +0100
+++ /tmp/kafka_enterprise_disabled.yaml	2021-12-14 20:51:50.000000000 +0100
@@ -241,24 +241,6 @@
         replicaId: "$2"
         memberType: "$3"
 ---
-# Source: cp-helm-charts/charts/cp-control-center/templates/service.yaml
-apiVersion: v1
-kind: Service
-metadata:
-  name: RELEASE-NAME-cp-control-center
-  labels:
-    app: cp-control-center
-    chart: cp-control-center-0.1.0
-    release: RELEASE-NAME
-    heritage: Helm
-spec:
-  ports:
-    - name: cc-http
-      port: 9021
-  selector:
-    app: cp-control-center
-    release: RELEASE-NAME
----
 # Source: cp-helm-charts/charts/cp-kafka-connect/templates/service.yaml
 apiVersion: v1
 kind: Service
@@ -420,59 +402,6 @@
     app: cp-zookeeper
     release: RELEASE-NAME
 ---
-# Source: cp-helm-charts/charts/cp-control-center/templates/deployment.yaml
-apiVersion: apps/v1
-kind: Deployment
-metadata:
-  name: RELEASE-NAME-cp-control-center
-  labels:
-    app: cp-control-center
-    chart: cp-control-center-0.1.0
-    release: RELEASE-NAME
-    heritage: Helm
-spec:
-  replicas: 1
-  selector:
-    matchLabels:
-      app: cp-control-center
-      release: RELEASE-NAME
-  template:
-    metadata:
-      labels:
-        app: cp-control-center
-        release: RELEASE-NAME
-      annotations:
-        prometheus.io/scrape: "true"
-        prometheus.io/port: "5556"
-    spec:
-      containers:
-        - name: cp-control-center
-          image: "confluentinc/cp-enterprise-control-center:6.2.1"
-          imagePullPolicy: IfNotPresent
-          ports:
-            - name: cc-http
-              containerPort: 9021
-              protocol: TCP
-          resources:
-            {}
-          env:
-            - name: CONTROL_CENTER_BOOTSTRAP_SERVERS
-              value: PLAINTEXT://RELEASE-NAME-cp-kafka-headless:9092
-            - name: CONTROL_CENTER_ZOOKEEPER_CONNECT
-              value:
-            - name: CONTROL_CENTER_CONNECT_CLUSTER
-              value: http://RELEASE-NAME-cp-kafka-connect:8083
-            - name: CONTROL_CENTER_KSQL_URL
-              value: http://RELEASE-NAME-cp-ksql-server:8088
-            - name: CONTROL_CENTER_KSQL_ADVERTISED_URL
-              value: http://RELEASE-NAME-cp-ksql-server:8088
-            - name: CONTROL_CENTER_SCHEMA_REGISTRY_URL
-              value: http://RELEASE-NAME-cp-schema-registry:8081
-            - name: KAFKA_HEAP_OPTS
-              value: "-Xms512M -Xmx512M"
-            - name: "CONTROL_CENTER_REPLICATION_FACTOR"
-              value: "3"
----
 # Source: cp-helm-charts/charts/cp-kafka-connect/templates/deployment.yaml
 apiVersion: apps/v1
 kind: Deployment
@@ -881,7 +810,7 @@
         - name: jmx-config
           mountPath: /etc/jmx-kafka
       - name: cp-kafka-broker
-        image: "confluentinc/cp-server:6.2.1"
+        image: "confluentinc/cp-kafka:6.2.1"
         imagePullPolicy: "IfNotPresent"
         securityContext:
           runAsUser: 0
@@ -915,10 +844,6 @@
           value: "RELEASE-NAME-cp-zookeeper-headless:2181"
         - name: KAFKA_LOG_DIRS
           value: "/opt/kafka/data-0/logs"
-        - name: KAFKA_METRIC_REPORTERS
-          value: "io.confluent.metrics.reporter.ConfluentMetricsReporter"
-        - name: CONFLUENT_METRICS_REPORTER_BOOTSTRAP_SERVERS
-          value: "PLAINTEXT://RELEASE-NAME-cp-kafka-headless:9092"
         - name: "KAFKA_LISTENER_SECURITY_PROTOCOL_MAP"
           value: "PLAINTEXT:PLAINTEXT,EXTERNAL:PLAINTEXT"
         - name: "KAFKA_OFFSETS_TOPIC_REPLICATION_FACTOR"
```

### CODE REVIEW

**Comment by redsk** [comment]
@gAmUssA

File: charts/cp-kafka/templates/statefulset.yaml [hunk 1]
```diff
@@ -83,7 +83,13 @@ spec:
           mountPath: /etc/jmx-kafka
       {{- end }}
       - name: {{ template "cp-kafka.name" . }}-broker
+        {{- if .Values.image }}
         image: "{{ .Values.image }}:{{ .Values.imageTag }}"
+        {{- else if .Values.enableEnterpriseFeatures }}
+        image: "confluentinc/cp-enterprise-kafka:{{ .Values.imageTag }}"
```
Comment (gAmUssA) [inline charts/cp-kafka/templates/statefulset.yaml]: cp-enterprise-kafka images are deprecated. use cp-server https://hub.docker.com/r/confluentinc/cp-server

File: values.yaml [hunk 2]
```diff
@@ -43,7 +43,7 @@ cp-zookeeper:
 cp-kafka:
   enabled: true
   brokers: 3
-  image: confluentinc/cp-enterprise-kafka
+  # image: confluentinc/cp-enterprise-kafka
```
Comment (gAmUssA) [inline values.yaml]: use cp-server instead https://hub.docker.com/r/confluentinc/cp-server

**Review by gAmUssA — CHANGES_REQUESTED** [review]

**Comment by redsk** [comment]
@gAmUssA changes made as requested

**Comment by gAmUssA** [comment]
kindly rebase, @redsk 
PR LGTM

**Comment by redsk** [comment]
@gAmUssA rebased. Also changed two existing occurrences of `cp-enterprise-kafka`

**Comment by redsk** [comment]
@gAmUssA do you think this pr is ready to be merged?

**Comment by gAmUssA** [comment]
@redsk testing... hang on

**Comment by gAmUssA** [comment]
@redsk could you rebase with master (need to rename to main)? I want to test GH Actions integration. Thank you

**Review by gAmUssA — APPROVED** [review]

**Comment by redsk** [comment]
@gAmUssA done

**Comment by redsk** [comment]
@gAmUssA should I do something regarding the lint error about the version bump?

**Comment by gAmUssA** [comment]
@redsk I need to figure out versioning schema and releasing automation. I will merge this. Thank you so much for your hard work.

**Pull request merged**